### ISSUE TITLE
update-state fails: System.MissingMethodException: Cannot find instance field/property/member name ValueAtKey

### ISSUE THREAD

**Issue by Saikyun** [arcadia-unity/Arcadia#327]
Steps to reproduce:
```
user=> (use 'arcadia.core)
nil
user=> (def c (object-named "Main Camera"))
#'user/c
user=> c
#<Main Camera (UnityEngine.GameObject)>
user=> (state+ c :hej "hej")
"hej"
user=> (state c :hej)
"hej"
user=> (update-state c :hej (constantly "monika"))
System.MissingMethodException: Cannot find instance field/property/member name ValueAtKey
  NO_CLASS.NO_METHOD
* arcadia/core/update-state.__interop_ValueAtKey23822
* arcadia/core/update-state
* user/eval
  clojure.lang.Compiler.eval (object)
  clojure.lang.Compiler.eval (object)
* clojure/core/eval
* arcadia/internal/socket-repl/game-thread-evalfnfn
```

**Comment by nasser** [comment]
Fixed in f0abd05

**Comment by Saikyun** [comment]
Got this error instead now:
```
user=> (use 'arcadia.core)
nil
user=> (state+ (object-named "Main Camera") :hej "hej")
"hej"
user=> (state (object-named "Main Camera") :hej)
"hej"
user=> (update-state (object-named "Main Camera") :hej (constantly "monika"))
System.ArgumentException: No implementation of method: :snapshot of protocol: #'arcadia.core/ISnapshotable found for class: String
* clojure/core/-cache-protocol-fn
* arcadia/core/fnG
* arcadia/core/update-state
* user/eval
  clojure.lang.Compiler.eval (object)
  clojure.lang.Compiler.eval (object)
* clojure/core/eval
* arcadia/internal/socket-repl/game-thread-evalfnfn
```

**Comment by timsgardner** [comment]
fixed in 41b027efd3e8a2500444460569c02fc1fed2582e
